codex/marketplace.md

```diff
@@ -0,0 +1,157 @@
+---
+slug: 
+title: CODEX-MARKETPLACE
+name: Codex Storage Marketplace
+status: raw
+tags: codex
+editor: 
+contributors:
+  
+---
+
+## Abstract
+
+This specification describes a method for Codex hosts and client nodes to participate in a storage marketplace. 
+The goal is to create a storage marketplace that promotes durability.
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data availability, 
+data persistence guarantees and node storage incentives.
+To reach this goal, a data availability and retrieval mechanism is needed.
+Support for light clients, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| Storage Nodes | A Codex node that provides storage services to the marketplace.|
+| Validator Nodes | A Codex node that collects, validates and submits proofs to reward or penalize other storage nodes or validator nodes. |
+| Regular Nodes | The main Codex client that interacts with other nodes to locate and retrieve data. Also considered to be an ephemeral node (light client) |
+| Slots | An agreement between storage nodes and regular nodes to store data |
+
+### Storage Request
+
+A regular node can request storage of data by opening a storage contract on the Codex blockchain. 
+The storage contract is opened in the marketplace via the blockchain, and 
+the requested data is split into blocks with erasure coding.
+Blocks of data are placed into slots to begin the request,
+see [slots](#slots) discussed below. 
```

Comment:
The client erasure codes the dataset and splits it into chunks. These chunks are available to download by storage providers who want to fill a slot.

A storage request is then created by submitting a transaction to the marketplace contract with the request parameters.